src/podcast_processor/podcast_processor.py

```diff
@@ -219,28 +222,70 @@ def classify(
                 podcast_topic=task.podcast_description,
                 transcript="\n".join(excerpts),
             )
-            identification = self.call_model(model, system_prompt, user_prompt)
-            with open(f"{target_dir}/identification.txt", "w") as f:
-                f.write(identification)
-            with open(f"{target_dir}/prompt.txt", "w") as f:
-                f.write(user_prompt)
-
-    def call_model(self, model: str, system_prompt: str, user_prompt: str) -> str:
-        # log the request
-        self.logger.info(f"Calling model: {model}")
-        response = self.client.chat.completions.create(
-            model=model,
-            messages=[
-                {"role": "system", "content": system_prompt},
-                {"role": "user", "content": user_prompt},
-            ],
-            max_tokens=self.config.openai_max_tokens,
-            timeout=self.config.openai_timeout,
-        )
 
-        content = response.choices[0].message.content
-        assert content is not None
-        return content
+            # Create a temporary file to indicate processing is in progress
+            with open(f"{target_dir}/.in_progress", "w") as f:
+                f.write("Processing")
+
+            try:
+                identification = self.call_model(model, system_prompt, user_prompt)
+                if identification:
+                    with open(f"{target_dir}/identification.txt", "w") as f:
```

Comment:
nit: maybe factor out a variable `identification_path = f"{target_dir}/identification.txt"`

Reply:
and similarly for some of these other variables. Could do it up on L201 where `target_dir` is defined.